src/Anexia.MathematicalProgram/Solve/IntegerLinearProgramSolver.cs

```diff
@@ -0,0 +1,145 @@
+    private SolverParameter SolverParameter { get; }
+    private ILPSolverType SolverType { get; }
+
+    public string ModelAsLpFormat() => Solver.ExportModelAsLpFormat(true);
+
+    public int NumberOfConstraints() => Solver.NumConstraints();
+
+    public int NumberOfVariables() => Solver.NumVariables();
+
+
+    public IntegerLinearProgramSolver SetSolverConfigurations(SolverParameter solverParameter)
+    {
+        _ = Solver.SetNumThreads((int)solverParameter.NumberOfThreads.Value);
+
+        if (!solverParameter.TimeLimitInMilliseconds.Equals(TimeLimitInMilliseconds.Unbounded))
+            Solver.SetTimeLimit(solverParameter.TimeLimitInMilliseconds.Value);
+
+        if (solverParameter.EnableSolverOutput.Value) Solver.EnableOutput();
+
+        return new IntegerLinearProgramSolver(Solver, solverParameter, SolverType);
+    }
+
+    public IntegerLinearProgramSolver AddIntegerVariable(IInterval interval, string variableName, out Variable variable)
+    {
+        variable = Solver.MakeIntVar(interval.LowerBound.Value, interval.UpperBound.Value, variableName);
+
+        return new IntegerLinearProgramSolver(Solver, SolverParameter, SolverType);
+    }
+
+    public IntegerLinearProgramSolver AddConstraints(Constraints constraints)
+    {
+        foreach (var constraint in constraints)
+        {
+            var interval = constraint.Interval;
+            var solverConstraint = Solver.MakeConstraint(interval.LowerBound.Value, interval.UpperBound.Value);
+
+            foreach (var term in constraint.Terms)
+                solverConstraint.SetCoefficient(term.Variable, term.Coefficient.Value);
+        }
+
+        return new IntegerLinearProgramSolver(Solver, SolverParameter, SolverType);
+    }
+
+    public IntegerLinearProgramSolver AddObjective(Terms terms, bool minimize) =>
+        AddObjective(terms, Constant.Zero, minimize);
+
+    public IntegerLinearProgramSolver AddObjective(Terms terms, Constant constant, bool minimize)
+    {
+        var objective = Solver.Objective();
+
+        foreach (var term in terms) objective.SetCoefficient(term.Variable, term.Coefficient.Value);
+
+        objective.SetOffset(constant.Value);
+
+        if (minimize)
+            objective.SetMinimization();
+        else
+            objective.SetMaximization();
+
+        return new IntegerLinearProgramSolver(Solver, SolverParameter, SolverType);
+    }
+
+    public SolverResult Solve()
+    {
+        try
+        {
+            using var parameter = new MPSolverParameters();
+
+            parameter.SetDoubleParam(MPSolverParameters.DoubleParam.RELATIVE_MIP_GAP,
+                SolverParameter.RelativeGap.Value);
+
+            var resultStatus = Solver.Solve(parameter);
+
+            return new ResultHandling(Solver).Handle(resultStatus);
+        }
+        catch (Exception e)
+        {
+            Console.WriteLine(e);
+
+            throw new MathematicalProgramException($"Error in ILP solver: {e.Message}, {e.InnerException}");
```

Comment:
Instead of printing it would be better to pass e to the MathematicalProgramException